pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -541,12 +547,23 @@ func isSecretValid(existingSecret v1.Secret) bool {
 	}
 
 	// if the calculated hash is different from the calculation, then it's invalid
-	if existingSecret.Annotations[esv1beta1.AnnotationDataHash] != utils.ObjectHash(existingSecret.Data) {
+	if existingSecret.Annotations[esv1beta1.AnnotationDataHash] != hashData(existingSecret.Data) {
 		return false
 	}
 	return true
 }
 
+func (r *Reconciler) setDataHashAnnotation(existing *v1.Secret, secret *v1.Secret) {
```

Comment:
nit, Removing side-effects makes it more readable. I would write 

```
func (r *Reconciler) computeDataHashAnnotation(existing *v1.Secret, secret *v1.Secret) string
```

and

```
secret.Annotations[esv1beta1.AnnotationDataHash] = r.computeDataHashAnnotation(&existingSecret, secret)
```

in the mutationFunc.